File: app/src/main/java/de/markusressel/typedpreferencesdemo/PreferenceHandler.java
package de.markusressel.typedpreferencesdemo;

import android.content.Context;
import android.support.annotation.NonNull;

import java.util.LinkedList;
import java.util.List;

import javax.inject.Inject;
import javax.inject.Singleton;

import de.markusressel.typedpreferences.PreferenceItem;
import de.markusressel.typedpreferences.PreferencesHandlerBase;

/**
 * Created by Markus on 18.07.2017.
 */
@Singleton
public class PreferenceHandler extends PreferencesHandlerBase {

    public static final PreferenceItem<Integer> THEME = new PreferenceItem<>(R.string.key_theme, 0);
    public static final PreferenceItem<Boolean> BOOLEAN_SETTING = new PreferenceItem<>(R.string.key_boolean_setting, true);

    private List<PreferenceItem> allPreferences;

    @Inject
    public PreferenceHandler(Context context) {
        super(context);

        allPreferences = new LinkedList<>();
        allPreferences.add(THEME);
        allPreferences.add(BOOLEAN_SETTING);
    }

    @NonNull
    @Override
    public String getSharedPreferencesName() {
        return "preferences";
    }

    @NonNull
    @Override
    public List<PreferenceItem> getAllPreferenceItems() {
        return allPreferences;
    }

}

File: app/src/main/java/de/markusressel/typedpreferencesdemo/PreferenceHandler.kt
/*
 * Copyright (c) 2017 Markus Ressel
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package de.markusressel.typedpreferencesdemo

import android.content.Context
import de.markusressel.typedpreferences.PreferenceItem
import de.markusressel.typedpreferences.PreferencesHandlerBase
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Markus on 18.07.2017.
 */
@Singleton
class PreferenceHandler @Inject
constructor(context: Context) : PreferencesHandlerBase(context) {

    // be sure to override the get() method
    override val sharedPreferencesName: String
        get() = "preferences"

    override val allPreferenceItems: Set<PreferenceItem<*>> = hashSetOf(
            THEME,
            BOOLEAN_SETTING,
            COMPLEX_SETTING
    )

    companion object {
        val THEME = PreferenceItem(R.string.key_theme, 0)
        val BOOLEAN_SETTING = PreferenceItem(R.string.key_boolean_setting, true)
        val COMPLEX_SETTING = PreferenceItem(R.string.key_complex_setting, ComplexClass("Complex ^", 10, listOf(1, 2, 3)))
    }

}
